### PR TITLE
chore(hardware-trezor): trezor tx certificates and auxiliaryData mapper 

### PR DESCRIPTION
**Pull request by tomislavhoracek**
This PR introduces the implementation of mapping Cardano tx certificates and auxiliaryData to Trezor certificates and auxiliaryData.

[Jira Ticket](https://input-output.atlassian.net/browse/LW-6800)

# TODO
- [x] Stake registration certificates
- [x] Stake deregistration certificates
- [x] Stake delegation certificates
- [x] Stake pool registration certificates
- [x] Auxiliary data mapper

### CODE REVIEW

File: packages/hardware-trezor/src/transformers/certificates.ts [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+  );
+  const rewardAddress = knownAddress ? Cardano.Address.fromBech32(knownAddress.rewardAccount)?.asReward() : null;
+
+  let credentials;
+  if (rewardAddress?.getPaymentCredential().type === Cardano.CredentialType.KeyHash) {
+    const path = stakeKeyPathFromGroupedAddress(knownAddress);
+    credentials = path ? { path } : { keyHash: certificate.stakeKeyHash };
+  } else {
+    credentials = { scriptHash: certificate.stakeKeyHash };
+  }
```
Comment (DominikGuzei) [inline packages/hardware-trezor/src/transformers/certificates.ts:71]: this conditional block to generate the credentials is duplicated between `getStakeAddressCertificate` and `getStakeDelegationCertificate`, maybe we could move that into a separate helper?

File: packages/hardware-trezor/src/transformers/certificates.ts [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+            return {
+              hostName: relay.hostname,
+              port: relay.port,
+              type: Trezor.CardanoPoolRelayType.SINGLE_HOST_NAME
+            };
+          case 'RelayByNameMultihost':
+            return {
+              hostName: relay.dnsName,
+              type: Trezor.CardanoPoolRelayType.MULTIPLE_HOST_NAME
+            };
```
Comment (DominikGuzei) [inline packages/hardware-trezor/src/transformers/certificates.ts:133]: is it ok to have no `default` fallback case here?

**Review by DominikGuzei — APPROVED** [review]
Great work @tomislavhoracek 👏 I just added two minor comments

**Comment by tomislavhoracek** [comment]
@AngelCastilloB @DominikGuzei Thanks guys for the review.
Changes you requested are addressed in [5f35deb](https://github.com/input-output-hk/cardano-js-sdk/pull/802/commits/5f35deb7870a4f61333f0f88de89c4be5e0f7b3f)

File: packages/hardware-trezor/src/transformers/certificates.ts [hunk 3]
```diff
@@ -0,0 +1,173 @@
...
+  stakeKeyHash: Crypto.Ed25519KeyHashHex,
+  knownAddresses: GroupedAddress[] | undefined
+): CertCredentialsType => {
+  const knownAddress = knownAddresses?.find(
+    (address) => Cardano.RewardAccount.toHash(address.rewardAccount) === stakeKeyHash
+  );
+  const rewardAddress = knownAddress ? Cardano.Address.fromBech32(knownAddress.rewardAccount)?.asReward() : null;
+
+  if (rewardAddress?.getPaymentCredential().type === Cardano.CredentialType.KeyHash) {
+    const path = stakeKeyPathFromGroupedAddress(knownAddress);
```
Comment (DominikGuzei) [inline packages/hardware-trezor/src/transformers/certificates.ts:57]: Im getting the following eslint error here:
grafik

Comment (tomislavhoracek) [reply packages/hardware-trezor/src/transformers/certificates.ts:57]: @DominikGuzei everything is green on my side and also the pre-commit hook didn't catch it.
Also, [stakeKeyPathFromGroupedAddress](https://github.com/input-output-hk/cardano-js-sdk/blob/chore/lw-6800-certificates-and-aux-mapper-for-trezor-sdk-implementation/packages/hardware-trezor/src/transformers/keyPaths.ts#L19) is accepting `GroupedAddress | undefined`

Comment (tomislavhoracek) [reply packages/hardware-trezor/src/transformers/certificates.ts:57]: You probably merged this branch into your miscellaneous mapper. Did you maybe change something in https://github.com/input-output-hk/cardano-js-sdk/blob/chore/lw-6800-certificates-and-aux-mapper-for-trezor-sdk-implementation/packages/hardware-trezor/src/transformers/keyPaths.ts?
Also, can you check it in this PR `chore/lw-6800-certificates-and-aux-mapper-for-trezor-sdk-implementation` branch ?

Comment (DominikGuzei) [reply packages/hardware-trezor/src/transformers/certificates.ts:57]: Hm interesting, I didn't change anything outside `tx.ts` 🤔

Comment (DominikGuzei) [reply packages/hardware-trezor/src/transformers/certificates.ts:57]: @tomislavhoracek all good, it was actually a caching issues of Webstorm 🙈 when i followed the type signature of the function it updated and the error disappeared 👍

Comment (tomislavhoracek) [reply packages/hardware-trezor/src/transformers/certificates.ts:57]: Thanks D ❤️

**Pull request merged**